website/docs: update docker outpost integration documentation for docker socket proxies

itskagee opened the pull request:
- Changed the docker socket proxy link to point to a better and more maintained solution.
- Changed permissions to include System/Info.
- Added a section on how to connect to a docker socket proxy.

1. The existing solution is sparsely maintained. This new docker socket proxy solution is an enhanced and hardened fork of the current one and is actively maintained by a reputed team.
2. System/Info permission is required, otherwise the outpost integration doesn't work.
3. Most documentation on docker socket proxies list examples which connect over TCP, but that doesn't work with authentik. This can be confusing to the average user. Hence, added a section to reflect that connections must be made over HTTP.

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

itskagee committed:
Update website docs for docker outpost integration

- Changed the docker socket proxy link to point to a better and more maintained solution.
- Changed permissions to include System/Info.
- Added a section on how to connect to a docker socket proxy.

Signed-off-by: itskagee <17406594+itskagee@users.noreply.github.com>

BeryJu committed:
add description for System Info permission

Signed-off-by: Jens Langhammer <jens@goauthentik.io>

tanberry commented on website/docs/add-secure-apps/outposts/integrations/docker.md:
```diff
@@ -32,13 +32,18 @@ The container is created with the following hardcoded properties:
 ## Permissions
 
-To minimise the potential risks of mapping the Docker socket into a container/giving an application access to the Docker API, many people use Projects like [docker-socket-proxy](https://github.com/Tecnativa/docker-socket-proxy). authentik requires these permissions from the Docker API:
+To minimise the potential risks of mapping the Docker socket into a container/giving an application access to the Docker API, many people use Projects like [docker-socket-proxy](https://docs.linuxserver.io/images/docker-socket-proxy/). authentik requires these permissions from the Docker API:
```
```suggestion
To minimise the potential risks of mapping the Docker socket to a container/giving an application access to the Docker API, many people use Projects like [docker-socket-proxy](https://docs.linuxserver.io/images/docker-socket-proxy/). authentik requires the following permissions from the Docker API:
```

itskagee replied:
Oh you're absolutely right.

I honestly just made the bare minimum changes possible (just the url in the paragraph above) because I was editing this on Github itself. Didn't want to bother with the lint checks and makes.

But since I've already cloned the repo for another PR I made, I'll make all the proper changes for this file too and make another commit soon.

tanberry commented on website/docs/add-secure-apps/outposts/integrations/docker.md:
```diff
@@ -32,13 +32,18 @@ The container is created with the following hardcoded properties:
+
+## Docker Socket Proxy
+
+Connections to a docker socket proxy must be made over `http`, not `tcp`, e.g. `http://<docker-socket-proxy hostname/container name>:<port>`.
```
```suggestion
Connections to a docker socket proxy must be made over HTTP, not TCP, e.g. `http://<docker-socket-proxy hostname/container name>:<port>`.
```

tanberry replied:
The way you had it is also correct, *if* those are the exact values shown in the UI, but it is standard to capitalize the names of protocols (and I think it's easier to read/skim).

itskagee replied:
That makes sense.

Thankyou for the suggestion!

tanberry reviewed (changes requested):
Thanks so much @itskagee for this PR! A few edits and questions, please.

itskagee committed:
Apply suggestions from code review

Co-authored-by: Tana M Berry <tanamarieberry@yahoo.com>
Signed-off-by: itskagee <17406594+itskagee@users.noreply.github.com>

itskagee committed:
website/docs: updated docker outpost integration for docker socket proxies

- Added additional information regarding docker socket proxies.
- Changed the layout to better present said information.

itskagee commented:
@tanberry I have merged some changes based on your suggestions.

I have also taken the liberty to move some information out of the 'permissions' section and into the 'docker socket proxy' section. It seemed more appropriate there.

Please let me know whether this is fine or if any further changes or enhancements are required!

itskagee committed:
Merge branch 'goauthentik:main' into patch-2

tanberry commented on website/docs/add-secure-apps/outposts/integrations/docker.md:
```diff
@@ -32,13 +32,26 @@ The container is created with the following hardcoded properties:
+
+See [permissions](#permissions) for the list of APIs that authentik needs access to.
+
+Note: Connections from authentik to docker socket proxy must be made over HTTP, not TCP, e.g. `http://<docker-socket-proxy hostname/container name>:<port>`.
```
```suggestion
Note: Connections from authentik to Docker socket proxy must be made over HTTP, not TCP, e.g. `http://<docker-socket-proxy hostname/container name>:<port>`.
```

tanberry committed:
Update website/docs/add-secure-apps/outposts/integrations/docker.md

Signed-off-by: Tana M Berry <tanamarieberry@yahoo.com>

tanberry reviewed (approved):
Those are great edits, thanks!
